### PR TITLE
Update readme for clarity

### PR DESCRIPTION
**Pull request by CeleritasCelery**
@DamienCassou 

Please Review.

### CODE REVIEW

File: README.org [hunk 1]
```diff
@@ -3,8 +3,12 @@
...
 [[https://github.com/CeleritasCelery/emacs-native-shell-complete/actions?query=workflow%3ACI][file:https://github.com/CeleritasCelery/emacs-native-shell-complete/workflows/CI/badge.svg]]
 
-This package interacts with a shell's native completion functionality to provide
-the same completions in Emacs that you would get from the shell itself.
+This package provides the exact same completions in ~shell-mode~ as you get in a
+terminal emulator with the =TAB= key. This means that completions are always
+accurate and synchronized with your current directory and shell environment.
+This also means that subshell completion works correctly. This is in contrast to
+packages like [[https://github.com/szermatt/emacs-bash-completion][emacs-bash-completion]] that use a separate process to generate
+"best guess" completions.
```
Comment (DamienCassou) [inline README.org:11]: awesomely clear!

File: README.org [hunk 2]
```diff
@@ -24,29 +27,48 @@ shell buffer. ~company-native-complete~ provides a function of the same name whi
...
+*** csh
+Make sure editing is enabled.
+#+BEGIN_SRC sh
+  set edit
+#+END_SRC
+
+** Advanced setup
+For most users this package should work out the box. However if you are using an
+unusual shell or are invoking subshells that are a different type then your main
+shell, you may need add some additional configuration.
```
Comment (DamienCassou) [inline README.org:53]: This is an important introduction. Thanks

**Review by DamienCassou — APPROVED** [review]
Great job, thank you! It really clarifies your README.

I'm missing a section explaining the differences between the different shells and terminals though.

**Comment by CeleritasCelery** [comment]
Thanks

**Pull request merged**